packages/toolkit/src/lib/react-query-service/mgmt/use-authenticated-user/server.ts

```diff
@@ -17,12 +17,58 @@ export async function fetchAuthenticatedUser({
 
     const user = await client.core.user.getAuthenticatedUser();
 
+    // Normalize avatar URL so it works in <img> (rewrite protected endpoints to local proxy)
+    if (user.profile && typeof user.profile.avatar === "string") {
+      const original = user.profile.avatar.trim();
+      user.profile.avatar = normalizeAvatarUrl(original);
+    }
+
     return Promise.resolve(user);
   } catch (error) {
     return Promise.reject(error);
   }
 }
 
+// Normalize avatar URL:
+// - data URLs: returned as-is
+// - versioned API endpoints /v1|v1alpha|v1beta/users/{id}/avatar -> rewritten to /api/avatar/{id} (keep query)
+// - http(s) or root-relative URLs: returned as-is
+// - otherwise: undefined (fallback)
+function normalizeAvatarUrl(url: string): string | undefined {
```

Comment:
This feels like an over-engineered way to get the avatar urls. Do we even have search query params in the avatar urls? I understand that always getting the url as `/avatar` without understanding whether it's an image or nothing is a bit confusing, but it's a simple logic. This update makes it look like we have some complex ways of getting the avatar urls, while in fact it stayed the same. I feel like this increases complexity of the code without any benefits. Maybe I'm missing something.